resources/group.go

```diff
@@ -225,6 +224,22 @@ func (obj *GroupRes) CheckApply(apply bool) (checkOK bool, err error) {
 type GroupUID struct {
 	BaseUID
 	name string
+	gid  *uint32
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false.
+func (obj *GroupUID) IFF(uid ResUID) bool {
+	res, ok := uid.(*GroupUID)
+	if !ok {
+		return false
+	}
+	var g, n bool
+	if obj.gid != nil && res.gid != nil {
+		g = *obj.gid == *res.gid
+	}
+	n = obj.name == res.name
+	// return true if gid or name are the same
+	return g || n
```

Comment:
I'm going to make a giant guess here: you probably settled on `g || n` instead of `g && n` to get it to work. But here's why the && (which would have been correct) wouldn't work in your testing, it's subtle: since we have some situations where we can only build an autoedge with a gid, (we have no knowledge in the user what name the group is) but we *always* know on the group side what the name is (that's the primary key) so the IFF function will in some scenarios be coming empty string to group name, and fail. So since an empty group name is invalid (please add that check for `group` and `groups` to Validate) we actually need to do a if `obj.name != "" && res.name != "" {  if .. != ... { return false }   }` similar to how we do the uid check?

In case none of this made sense, then maybe I'm just wrong for lack of sleep and failure to remember how my autoedges code works :P